reconcile/aws_version_sync/merge_request_manager/merge_request_manager.py

```diff
@@ -54,7 +48,7 @@ def process(self, gitlab_cli: GitLabApi) -> None:
         )
 
 
-class MergeRequestManager:
+class MergeRequestManager(MergeRequestManagerBase):
```

Comment:
The usage of `MergeRequestManager` not match inheritance. If use inheritance, `MergeRequestManagerBase` should define all interface method that be used in integration. Like this:

```python
class MergeRequestManagerBase:
    def housekeeping(self): ...

    @abstractmethod
    def expected_data(self):
        pass

    def _merge_request_already_exists(self, expected_data): ...

    def create_merge_request(self):
        data = self.expected_data()
        if self._merge_request_already_exists(data):
            ...
        ...
```

each subclass implement abstract method, integration should only use public interface of base class, not methods in subclass.

If there are too many different creation logic in subclass, the whole `create_merge_request` can be abstract, but in that case we should use composition over inheritance, change `is a` to `has a`, each `MergeRequestManager` has a `MergeRequestManagerUtil` do common tasks like check mr exist used in it's own create method like `create_avs_merge_request`.

Reply:
made create merge request an abstract method